MinecraftClient.py

```diff
@@ -0,0 +1,180 @@
+import argparse
+import os, sys
+import re
+import atexit
+from subprocess import Popen
+from shutil import copyfile
+from base64 import b64decode
+from time import strftime
+
+import requests
+
+import Utils
+
+atexit.register(input, "Press enter to exit.")
+
+# 1 or more digits followed by m or g, then optional b
+max_heap_re = re.compile(r"^\d+[mMgG][bB]?$")
+
+
+def prompt_yes_no(prompt):
+    yes_inputs = {'yes', 'ye', 'y'}
+    no_inputs = {'no', 'n'}
+    while True:
+        choice = input(prompt + " [y/n] ").lower()
+        if choice in yes_inputs: 
+            return True
+        elif choice in no_inputs: 
+            return False
+        else:
+            print('Please respond with "y" or "n".')
+
+
+# Find Forge jar file; raise error if not found 
+def find_forge_jar(forge_dir):
+    for entry in os.scandir(forge_dir):
+        if ".jar" in entry.name and "forge" in entry.name:
+            print(f"Found forge .jar: {entry.name}")
+            return entry.name
+    raise FileNotFoundError(f"Could not find forge .jar in {forge_dir}.")
+
+
+# Create mods folder if needed; find AP randomizer jar; return None if not found.
+def find_ap_randomizer_jar(forge_dir):
+    mods_dir = os.path.join(forge_dir, 'mods')
+    if os.path.isdir(mods_dir):
+        ap_mod_re = re.compile(r"^aprandomizer-[\d\.]+\.jar$")
+        for entry in os.scandir(mods_dir):
+            match = ap_mod_re.match(entry.name)
+            if match:
+                print(f"Found AP randomizer mod: {match.group()}")
+                return match.group()
+        return None
+    else:
+        os.mkdir(mods_dir)
+        print(f"Created mods folder in {forge_dir}")
+        return None
+
+
+# Create APData folder if needed; clean .apmc files from APData; copy given .apmc into directory.
+def replace_apmc_files(forge_dir, apmc_file):
+    if apmc_file is None:
+        return
+    apdata_dir = os.path.join(forge_dir, 'APData')
+    if not os.path.isdir(apdata_dir):
+        os.mkdir(apdata_dir)
+        print(f"Created APData folder in {forge_dir}")
+    for entry in os.scandir(apdata_dir):
+        if ".apmc" in entry.name and entry.is_file():
+            os.remove(entry.path)
+        print(f"Removed existing .apmc files in {apdata_dir}")
```

Comment:
This will print for every file in the APData folder (3 cwith current minecraft client) even though it's only removing one file.

Reply:
Fixed.